services/wallet/collectibles/collection_data_db.go

```diff
@@ -0,0 +1,145 @@
+package collectibles
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/status-im/status-go/services/wallet/thirdparty"
+)
+
+type CollectionDataDB struct {
+	db *sql.DB
+}
+
+func NewCollectionDataDB(sqlDb *sql.DB) *CollectionDataDB {
+	return &CollectionDataDB{
+		db: sqlDb,
+	}
+}
+
+const collectionDataColumns = "chain_id, contract_address, provider, name, slug, image_url, traits"
+
+func insertOrReplaceCollectionsData(creator statementCreator, collections []thirdparty.CollectionData) error {
+	insertCollection, err := creator.Prepare(fmt.Sprintf(`INSERT OR REPLACE INTO collection_data_cache (%s) 
+																				VALUES (?, ?, ?, ?, ?, ?, ?)`, collectionDataColumns))
+	if err != nil {
+		return err
+	}
+
+	for _, c := range collections {
+		_, err = insertCollection.Exec(
+			c.ID.ChainID,
+			c.ID.Address,
+			c.Provider,
+			c.Name,
+			c.Slug,
+			c.ImageURL,
+			&JSONBlob{c.Traits},
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *CollectionDataDB) SetData(collections []thirdparty.CollectionData) (err error) {
+	var (
+		tx *sql.Tx
+	)
+	tx, err = o.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		_ = tx.Rollback()
+	}()
+
+	// Insert new collections data
+	err = insertOrReplaceCollectionsData(tx, collections)
+	if err != nil {
+		return err
+	}
+
+	return
+}
+
+func scanCollectionsDataRow(row *sql.Row) (*thirdparty.CollectionData, error) {
+	c := thirdparty.CollectionData{
+		Traits: make(map[string]thirdparty.CollectionTrait),
+	}
+	err := row.Scan(
+		&c.ID.ChainID,
+		&c.ID.Address,
+		&c.Provider,
+		&c.Name,
+		&c.Slug,
+		&c.ImageURL,
+		&JSONBlob{&c.Traits},
```

Comment:
No way to split that into separate columns or linked tables? Or is there no need for future inquiry into data?

Reply:
Separate columns won't do, since each collectible/collection could own any number of traits. I could move this data to a separate table though. Don't think it's really needed unless in the future we want to filter collectibles by some trait (unlikely), though it's probably better for table data inspection to have it that way.

Reply:
Done!